Plugin.php

```diff
@@ -184,6 +186,28 @@ public function registerFormWidgets()
         ];
     }
 
+    public function localeUrl($url, $locale)
+    {
+        $translator = Translator::instance();
+        $parts = parse_url($url);
+        $path = array_get($parts, 'path');
+        $parts['path'] = '/' . $translator->getPathInLocale($path, $locale);
+        return $this->build_url($parts);
+    }
+
+    public function build_url(array $parts) {
```

Comment:
@mjauvin The method name should be in camel-case :)
```suggestion
    public function buildUrl(array $parts) {
```